src/query/graphite/native/aggregation_functions.go

```diff
@@ -278,6 +277,82 @@ func combineSeriesWithWildcards(
 	return r, nil
 }
 
+/*
+applyByNode takes a seriesList and applies some complicated function (described by a string), replacing templates with unique
+prefixes of keys from the seriesList (the key is all nodes up to the index given as `nodeNum`).
+
+If the `newName` parameter is provided, the name of the resulting series will be given by that parameter, with any
+"%" characters replaced by the unique prefix.
+
+Example:
+
+.. code-block:: none
+
+&target=applyByNode(servers.*.disk.bytes_free,1,"divideSeries(%.disk.bytes_free,sumSeries(%.disk.bytes_*))")
+
+Would find all series which match `servers.*.disk.bytes_free`, then trim them down to unique series up to the node
+given by nodeNum, then fill them into the template function provided (replacing % by the prefixes).
+
+Additional Examples:
+
+Given keys of
+
+- `stats.counts.haproxy.web.2XX`
+- `stats.counts.haproxy.web.3XX`
+- `stats.counts.haproxy.web.5XX`
+- `stats.counts.haproxy.microservice.2XX`
+- `stats.counts.haproxy.microservice.3XX`
+- `stats.counts.haproxy.microservice.5XX`
+
+The following will return the rate of 5XX's per service:
+
+.. code-block:: none
+
+applyByNode(stats.counts.haproxy.*.*XX, 3, "asPercent(%.5XX, sumSeries(%.*XX))", "%.pct_5XX")
+
+The output series would have keys `stats.counts.haproxy.web.pct_5XX` and `stats.counts.haproxy.microservice.pct_5XX`.
+*/
+func applyByNode(ctx *common.Context, seriesList singlePathSpec, nodeNum int, templateFunction string, newName string) (ts.SeriesList, error) {
+	// using this as a set
+	prefixMap := make(map[string]int)
```

Comment:
Better to use `map[string]struct{}` since it will not need to allocate

Reply:
I used `map[string]int{}` cause I need the values to be ints. Is that OK?

Reply:
So for maps that only check membership, like you have here, you typically just use `struct{}` as the map type, to avoid allocating the additional byte; then just check for key existence, e.g. `_, exists := prefixMap["check"]`, or range over only the keys, like you have here. This is a ["go-ish" optimization](https://gist.github.com/davecheney/3be245c92b61e5045f75#gistcomment-2671632) Typically the actual savings aren't huge, but we still want to try and keep allocs down as much as possible

Reply:
Ok, makes sense. I changed it to do it like this:

```
prefixMap := map[string]struct{}{}
```